modules/web_scraper.py

```diff
@@ -0,0 +1,238 @@
+    """
+    Extracts the data-url attributes from a tweet element.
+    """
+    # Getting all <a>-tags
+    tweet_a_tags = tweet_element.find_all("a")
+    data_urls = []
+    for element in tweet_a_tags:
+        try:
+            # If the tag has an attribute "data-url" then we take the link from this URL
+            data_url = element["data-url"]
+            data_urls.append(data_url)
+        except:
+            pass
+    return data_urls
+
+
+def _get_tweet_date(tweet_element):
+    """
+    Returns a datetime object with the date contained in the tweet.\nThe minutes and hours etc. are not accurate.\n
+    If the tweet is from before 2020 it will still be returned as if it was from 2020 :Hide_the_pain_Harold: Who needs tweets from the previous year anyway
+    """
+    # Finding element that contains the timestamp
+    timestamp_element = tweet_element.find("td", class_=["timestamp"])
+    # Inner element with timestamp or date
+    timestamp_inner_element = timestamp_element.find("a")
+    # The date is stored as inner text
+    timestamp = timestamp_inner_element.text
+    # If the timestamp contains "min" or "h"(hour) etc. it means the tweet is from today so we just return the date today
+    if "min" in timestamp or timestamp[-1] == "m" or timestamp[-1] == "s" or "h" in timestamp or "now" in timestamp:
+        return "{},{},{}".format(datetime.now().date().year, datetime.now().date().month, datetime.now().date().day)
+    # Extract date to ensure proper formatting
+    tweet_date = datetime.strptime(timestamp + " 2020", "%b %d %Y")
+    # The weird looking return statement is made to make it easier for other parts of the program to read the date
+    return "{},{},{}".format(tweet_date.date().year, tweet_date.date().month, tweet_date.date().day)
+
+
+def _emoji_description_extractor(text: str):
+    """ 58 lol
+    Used to extract the decription of emojis used in the given text.
+    This function only supports 842 emojis. Some descriptions may be empty if the used emoji is rare or whatever
+    """
+    emoji_list = emoji.emoji_lis(text)
+    emoji_descriptions = [str.strip(emoji.demojize(vars.get("emoji")).replace("_", " ").replace(":", "")) for vars in emoji_list]
+    return emoji_descriptions
+
+
+def _get_next_page_link(soup: bs4.BeautifulSoup):
+    """
+    Used to get the link to the next twitter page with older tweets. Extracts the link from given soup (bs4).
+    """
+    more_element = soup.find("div", class_=["w-button-more"])
+    a_tag = more_element.find("a")
+    next_page_link = a_tag["href"]
+    url_part = "https://mobile.twitter.com"
+    return url_part + next_page_link
+
+
+def get_tweets(tweet_count: int, fresh_search: bool, *hashtags: str):
+    """
+    --- Returns ---\n
+    List of strings. The contents of each string will be structured as a dictionary
+
+    --- Parameters ---\n
+    tweet_count: integer - The amount of tweets you want. Provide a number in an interval of 20.\n
```

Comment:
If this may ONLY be 20, 40, 60 etc, maybe there should be a check/guard for it? 
Make a check a la: `if tweet_count modulo 20 is not 0 raise Exception("Tweet has to be divisible by 20")`

Reply:
I see below, that providing 39 will result in 20, and not a crash / something else bad.

Reply:
Yeah it will always result in a minimum of 20